perun-core/src/main/java/cz/metacentrum/perun/core/impl/FacilitiesManagerImpl.java

```diff
@@ -449,6 +449,44 @@ public void removeOwner(PerunSession sess, Facility facility, Owner owner) throw
 		}
 	}
 
+	@Override
+	public List<Facility> getAllowedFacilities(PerunSession sess, User user) throws InternalErrorException {
+		try  {
+			return jdbc.query("select distinct " + facilityMappingSelectQuery + " from groups_resources" +
+							" join groups on groups_resources.group_id=groups.id" +
+							" join groups_members on groups.id=groups_members.group_id" +
+							" join members on groups_members.member_id=members.id " +
+							" join resources on groups_resources.resource_id=resources.id " +
+							" join facilities on resources.facility_id=facilities.id " +
+							" where members.user_id=? and members.status!=? and members.status!=?",
+					FACILITY_MAPPER, user.getId(),
+					String.valueOf(Status.INVALID.getCode()), String.valueOf(Status.DISABLED.getCode()));
+		} catch (EmptyResultDataAccessException e) {
```

Comment:
This exception is probably not thrown for method jdbc.query.

Reply:
Again you are right, we should clean up the code everywhere. Here fixed.